client-v2/src/bundles/frontsUIBundle.ts

```diff
@@ -220,17 +219,37 @@ const selectPriority = (
   { priority }: { priority: string }
 ): string => priority;
 
-const selectEditorFrontsByPriority = createSelector(
-  getFronts,
-  selectEditorFrontIds,
-  selectPriority,
-  (fronts, openFrontIds, priority) => {
-    return openFrontIds.filter(frontId => {
-      const frontConfig = fronts[frontId];
-      return frontConfig && frontConfig.priority === priority;
-    });
-  }
-);
+const createSelectEditorFrontsByPriority = () =>
+  createSelector(
+    getFronts,
+    selectEditorFrontIds,
+    selectPriority,
+    (fronts, frontIdsByPriority, priority) => {
+      const openFrontIds = frontIdsByPriority[priority] || [];
+      return compact(openFrontIds.map(frontId => fronts[frontId]));
+    }
+  );
+
+const createSelectFrontIdAndOpenStateByPriority = () => {
```

Comment:
A new selector for the FrontsList, which needed a refactor anyway because of the priority work.